var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -383,6 +383,35 @@ def configure(self):
                   '-{0}gtk'.format('' if '+gtk' in self.spec else 'no-'),
                   *config_args)
 
+    @when('@5.8:')
+    def configure(self):
+        config_args = self.common_config_args
+
+        if not sys.platform == 'darwin':
+            config_args.extend([
+                '-qt-xcb',
+            ])
+
+        if '~webkit' in self.spec:
+            config_args.extend([
+                '-skip', 'webengine',
+            ])
+
+        if '~opengl' in self.spec and self.spec.satisfies('@5.10:'):
+            config_args.extend([
+                '-skip', 'webglplugin',
+            ])
+
+        # relies on a system installed wayland, i.e. no spack package yet
+        # https://wayland.freedesktop.org/ubuntu16.04.html
+        # https://wiki.qt.io/QtWayland
+        config_args.extend(['-skip', 'wayland'])
+
+        configure('-no-eglfs',
+                  '-no-directfb',
+                  '-{0}gtk'.format('' if '+gtk' in self.spec else 'no-'),
+                  *config_args)
```

Comment:
Most of this is duplicated. Can you just add an if-statement to the old configure function?

Reply:
I thought about that, but figured, that the rest of the configure functions would need to be cleaned up too. And I really wasn't in the mood for that :D 

If you want this to be cleaned up, then I'd say we should do the other ones too. But that would require at least a couple of qt builds and some time from me. I may be able to get to this towards the end of June (i.e. I'll put it on my todo, but the next few weeks will be kind of busy)

I'd prefer this to be merged right away (because then it's done and I have no chance of forgetting it), but you can also leave it open and I'll try to pick up here again, once I find the time

Reply:
> I thought about that, but figured, that the rest of the configure functions would need to be cleaned up too.

Never open up pandora's box.

I would be fine with simply adding an if-statement and pretending we didn't see the other `configure` functions if you're okay with that.

Reply:
done although I think this will be coming back to greet us :)